From 4bfad6f2091e43871137c944cc02c380cf6a8495 Mon Sep 17 00:00:00 2001
From: Oliver Scott <olivercscott@gmail.com>
Date: Thu, 30 Jun 2022 13:47:42 -0400
Subject: [PATCH 38/38] Enable backup service by default for all users

* Useful for work profiles and secondary users

cdesai: Port to 14
Issue: calyxos#1057
Change-Id: I3b2e1797c2e0d9e3af477f624c7d3ab613c2ea2c
---
 .../server/backup/BackupManagerService.java       | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/services/backup/java/com/android/server/backup/BackupManagerService.java b/services/backup/java/com/android/server/backup/BackupManagerService.java
index ce9cdc2ae..bbf9aaa16 100644
--- a/services/backup/java/com/android/server/backup/BackupManagerService.java
+++ b/services/backup/java/com/android/server/backup/BackupManagerService.java
@@ -319,14 +319,7 @@ public class BackupManagerService extends IBackupManager.Stub {
     /**
      * If there is a "suppress" file for the system user, backup is inactive for ALL users.
      *
-     * Otherwise, the below logic applies:
-     *
-     * For the {@link #mDefaultBackupUserId}, backup is active by default. Backup is only
-     * deactivated if there exists a "suppress" file for the user, which can be created by calling
-     * {@link #setBackupServiceActive}.
-     *
-     * For non-main users, backup is only active if there exists an "activated" file for the user,
-     * which can also be created by calling {@link #setBackupServiceActive}.
+     * Otherwise, it is activated for all users.
      */
     private boolean isBackupActivatedForUser(int userId) {
         if (getSuppressFileForUser(UserHandle.USER_SYSTEM).exists()) {
@@ -341,11 +334,7 @@ public class BackupManagerService extends IBackupManager.Stub {
             return false;
         }
 
-        if (isDefaultUser && getSuppressFileForUser(userId).exists()) {
-            return false;
-        }
-
-        return isDefaultUser || getActivatedFileForUser(userId).exists();
+        return true;
     }
 
     @VisibleForTesting
-- 
2.47.0

